### PR TITLE
Build tests and samples during regular build

### PR DESCRIPTION
**Pull request by victorhurdugaci**
cc @pranavkm @troydai @BrennanConroy

### CODE REVIEW

File: build/_k-standard-goals.shade [hunk 1]
```diff
@@ -96,7 +96,7 @@ default DNX_TOOLS_FEED = 'https://www.myget.org/F/dnxtools/api/v2'
   @{
     if (IsBuildV2) 
     {
-        const string buildPattern = "src/**";
+        const string buildPattern = "src/**;test/**;samples/**";
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: The way we do publishing \ Coherence builds now is we grab everything in the artifacts directory. Now that we'll have tests and samples copied to artifacts (https://github.com/aspnet/Universe/pull/258/files#diff-4405a98bc7f6daec30b9a687f626cbd4R115) is there a way to filter them out of being published to the feed? Should we change the structure of packages so `src` -> `artifacts/build`, `test` -> `artifacts/test`, `samples` -> `artifacts/samples`?

**Comment by pranavkm** [comment]
:watch:

**Comment by victorhurdugaci** [comment]
Updated

**Comment by BrennanConroy** [comment]
:sheep: it

**Comment by pranavkm** [comment]
:shipit:

File: build/_k-standard-goals.shade [hunk 2]
```diff
@@ -118,6 +110,38 @@ default DNX_TOOLS_FEED = 'https://www.myget.org/F/dnxtools/api/v2'
     }
   }
 
+#build-compile target='compile' if='IsBuildV2'
```
Comment (bricelam) [inline build/_k-standard-goals.shade]: I know it doesn't really matter, but should this run as part of `test-compile`instead?

Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: It doesn't matter :)

Comment (bricelam) [inline build/_k-standard-goals.shade]: I just say it because I often run `build compile` to get nupkgs as quickly as possible.

**Pull request merged**